yt/yt/server/scheduler/scheduler.cpp

```diff
@@ -2659,6 +2660,31 @@ class TScheduler::TImpl
             return;
         }
 
+        if (operation->Spec()->IssueTemporaryToken) {
+            try {
+                // We check this first, so that we don't create a token in vain.
+                auto secureVault = operation->GetSecureVault();
+                if (secureVault && secureVault->FindChild(operation->Spec()->TemporaryTokenEnvironmentVariableName)) {
+                    THROW_ERROR_EXCEPTION(
```

Comment:
I think it is better to check this before (in `StartOperation` or even inside spec postprocessing)

Reply:
Moved into spec postprocessing, it is much better, thanks. We also have extra environment variable name validation that I missed this way.